### ISSUE TITLE
compiling warns of using deprecated NumPy API

### ISSUE THREAD

**Issue by orbeckst** [MDAnalysis/mdanalysis#2988]
## Expected behavior ##

Compiling the MDA C extensions does not raise warnings about deprecated APIs.

## Actual behavior ##

Compiling the cython extension warns

*"Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"*.

This is due to Cython <3.0 using the deprecated NumPp API. This will change with upcoming Cython 3.0, see https://cython.readthedocs.io/en/latest/src/userguide/migrating_to_cy30.html#numpy-c-api at which point we can simply 
```python
define_macros.append("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION"))
```
in `setup.py`.

(However, with Cython < 3.0, compilation fails as it relies on the old API.)

## Code to reproduce the behavior ##

```bash
python setup.py buil
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.0.0-dev 
- Which version of Python (`python -V`)? 3.7
- Which operating system? macOS
- Cython: 0.29.21
